### DISCUSSION TITLE
Disabling ID in Document Payload

### DISCUSSION THREAD

**Discussion by james-a-roland** [wovalle/fireorm#280] [Q&A] [answered]
Hi there! I noticed that the ID is written by default to the document - is there some annotation marker to disable this? Looked through the tests and didn't see anything that popped out. Thanks!

**Comment by wovalle** [comment, marked as answer]
As of now this is a design choice/limitation. More info in https://github.com/wovalle/fireorm/issues/207

**Reply by james-a-roland** [reply]
Thanks for the quick response @wovalle! I tried using the [@Ignore](https://github.com/wovalle/fireorm/blob/master/src/Decorators/Ignore.ts) decorator on the id field, which seemed to work. 
- On the write path, this just deletes the K/V pair in the serialized dictionary. 
- On the read path, there's no key K/V pair that was written, so no pair to to read,

Does that sound correct, or am I missing something?

**Comment by wovalle** [comment]
Hey James! That's unsupported from the library point of view but I don't
see why it wouldn't work

On Mon, Oct 18, 2021 at 5:04 PM James Roland ***@***.***>
wrote:

> Thanks for the quick response @wovalle <https://github.com/wovalle>! I
> tried using the @Ignore
> <https://github.com/wovalle/fireorm/blob/master/src/Decorators/Ignore.ts>
> decorator on the id field, which seemed to work.
>
>    - On the write path, this just deletes the K/V pair in the serialized
>    dictionary.
>    - On the read path, there's no key K/V pair that was written, so no
>    pair to to read,
>
> Does that sound correct, or am I missing something?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wovalle/fireorm/discussions/280#discussioncomment-1496649>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB35QJD7BX42WJWKI2UGLUDUHQZPVANCNFSM5GFVT3KQ>
> .
>


-- 


Willy Ovalle
